test/models/orderbook.spec.ts

```diff
@@ -0,0 +1,31 @@
+import {Offer, Orderbook, Price} from "../../src/orderbook";
+import {assert} from "chai";
+import "mocha";
+
+describe("Orderbook", () => {
+  it("cummulates bids and asks sorted by best bid/best ask", () => {
+    const orderbook = new Orderbook("USDC", "DAI");
+    orderbook.addAsk(new Offer(new Price(11, 10), 100));
+    orderbook.addAsk(new Offer(new Price(12, 10), 200));
+    orderbook.addAsk(new Offer(new Price(101, 100), 300));
+
+    orderbook.addBid(new Offer(new Price(9, 10), 50));
+    orderbook.addBid(new Offer(new Price(99, 100), 70));
+    orderbook.addBid(new Offer(new Price(9, 10), 30));
+
+    assert.equal(
+      JSON.stringify(orderbook.toJSON()),
+      JSON.stringify({
+        bids: [
+          {price: 0.99, volume: 70},
+          {price: 0.9, volume: 80}
```

Comment:
I see that identical bids are aggregated by accumulating the volume. I suppose this was the reason for using the fraction class (so that prices could be in a reduced format). In this case we might want to consider including a unit test with `Price(F_n, F_{n-1})` where the F_i's are Fibonacci numbers and `n` is large enough so that `F_n` is as close as possible to the largest number in our system (i.e. maxU256), because this will yield the longest possible run time for computing the GCD.